src/predictor/gpu_predictor.cu

```diff
@@ -324,18 +325,20 @@ class GPUPredictor : public xgboost::Predictor {
     }
 
     size_t i_batch = 0;
-
+    size_t batch_offset = 0;
     for (const auto &batch : dmat->GetRowBatches()) {
-      CHECK_EQ(i_batch, 0) << "External memory not supported";
+      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
       // out_preds have been sharded and resized in InitOutPredictions()
-      batch.offset.Shard(GPUDistribution::Overlap(devices_, 1));
+      batch.offset.Reshard(GPUDistribution::Overlap(devices_, 1));
```

Comment:
Why do we need `Reshard()` here? Won't the distribution be the same as long as `devices_` stays the same?

Reply:
Done.